### DISCUSSION TITLE
Handle Request and Form

### DISCUSSION THREAD

**Discussion by laurentmuller** [symfony/symfony#44078] [Q&A]
### Description

In a controller, we use several time a code below:

```php
public function edit(Request $request, Conference $conference): Response
{
    $form = $this->createForm(ConferenceType::class, $conference);

    $form->handleRequest($request);
    if ($form->isSubmitted() && $form->isValid()) {
       ...
    }
    ...
}
```

### Example

It is not possible to create a function in the AbstractController to handle the request AND the form with code like this:

```php
protected function handleRequestForm(Request $request, FormInterface $form): bool  {

    $form->handleRequest($request);

    return $form->isSubmitted() && $form->isValid();
}
```

So the code is more clean:

```php
public function edit(Request $request, Conference $conference): Response
{
    $form = $this->createForm(ConferenceType::class, $conference);

    if ($this->handleRequestForm($request, $form)) {
       ...
    }
    ...
}
```

**Comment by micheh** [comment]
In version 5.3 a new helper method was added to ease with form submissions, where you could call `handleForm` and provide a callable which was executed when the form was submitted and valid (like you suggest): #40799

This was later reverted and will not be included in version 5.4. See #41178 for the discussion about the removal of the newly introduced helper method `handleForm`.
